docs/k0sctl-install.md

```diff
@@ -0,0 +1,177 @@
+   * Drain can be skipped with a --no-drain option
+
+
+In the k0sctl configuration, change the value of `spec.k0s.version`:
+```yaml
+  k0s:
+    version: 0.11.0
+```
+
+```sh
+$ k0sctl apply
+
+⠀⣿⣿⡇⠀⠀⢀⣴⣾⣿⠟⠁⢸⣿⣿⣿⣿⣿⣿⣿⡿⠛⠁⠀⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠀█████████ █████████ ███
+⠀⣿⣿⡇⣠⣶⣿⡿⠋⠀⠀⠀⢸⣿⡇⠀⠀⠀⣠⠀⠀⢀⣠⡆⢸⣿⣿⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀███          ███    ███
+⠀⣿⣿⣿⣿⣟⠋⠀⠀⠀⠀⠀⢸⣿⡇⠀⢰⣾⣿⠀⠀⣿⣿⡇⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠀███          ███    ███
+⠀⣿⣿⡏⠻⣿⣷⣤⡀⠀⠀⠀⠸⠛⠁⠀⠸⠋⠁⠀⠀⣿⣿⡇⠈⠉⠉⠉⠉⠉⠉⠉⠉⢹⣿⣿⠀███          ███    ███
+⠀⣿⣿⡇⠀⠀⠙⢿⣿⣦⣀⠀⠀⠀⣠⣶⣶⣶⣶⣶⣶⣿⣿⡇⢰⣶⣶⣶⣶⣶⣶⣶⣶⣾⣿⣿⠀█████████    ███    ██████████
+
+INFO[0000] k0sctl 0.5.0 Copyright 2021, Mirantis Inc.   
+INFO[0000] Anonymized telemetry will be sent to Mirantis. 
+INFO[0000] By continuing to use k0sctl you agree to these terms: 
+INFO[0000] https://k0sproject.io/licenses/eula          
+INFO[0000] ==> Running phase: Connect to hosts                   
+INFO[0000] [ssh] 10.0.0.17:22: connected              
+INFO[0000] [ssh] 10.0.0.23:22: connected            
+INFO[0000] ==> Running phase: Detect host operating systems 
+INFO[0000] [ssh] 10.0.0.23:22: is running CentOS Linux 7 (Core) 
+INFO[0000] [ssh] 10.0.0.17:22: is running CentOS Linux 7 (Core) 
+INFO[0000] ==> Running phase: Prepare hosts   
+INFO[0000] ==> Running phase: Gather host facts 
+INFO[0000] [ssh] 10.0.0.23:22: discovered bond0 as private interface      valid_lft forever preferred_lft forever 
+INFO[0000] [ssh] 10.0.0.23:22: discovered 10.12.18.133 as private address 
+INFO[0000] ==> Running phase: Validate hosts   
+INFO[0000] ==> Running phase: Gather k0s facts 
+INFO[0000] [ssh] 10.0.0.23:22: found existing configuration                 
+INFO[0000] [ssh] 10.0.0.23:22: is running k0s controller version 0.10.0 
+INFO[0000] [ssh] 10.0.0.23:22: need upgrade: true                     
+INFO[0000] [ssh] 10.0.0.17:22: is running k0s worker version 0.10.0 
+INFO[0000] [ssh] 10.0.0.17:22: need upgrade: true   
+INFO[0000] [ssh] 10.0.0.23:22: checking if worker wrk has joined 
+INFO[0001] ==> Running phase: Configure K0s    
+INFO[0001] ==> Running phase: Upgrade controllers 
+INFO[0001] [ssh] 10.0.0.23:22: starting upgrade
+INFO[0001] [ssh] 10.0.0.23:22: Running with legacy service name, migrating... 
+INFO[0011] [ssh] 10.0.0.23:22: waiting for the k0s service to start 
+INFO[0016] ==> Running phase: Upgrade workers  
+INFO[0016] Upgrading 1 workers in parallel              
+INFO[0016] [ssh] 10.0.0.17:22: upgrade starting  
+INFO[0027] [ssh] 10.0.0.17:22: waiting for node to become ready again 
+INFO[0027] [ssh] 10.0.0.17:22: upgrade successful   
+INFO[0027] ==> Running phase: Disconnect from hosts 
+INFO[0027] ==> Finished in 27s                 
+INFO[0027] k0s cluster version 0.11.0 is now installed 
+INFO[0027] Tip: To access the cluster you can now fetch the admin kubeconfig using: 
+INFO[0027]      k0sctl kubeconfig 
+```
+
+
+### Known limitations
+
+* k0sctl is not able to detect existing setup e.g. number of cluster nodes and similar
```

Comment:
Not sure what this means - does this mean that k0sctl is only aware of nodes that it installed? 
Sorry, something about the phrasing isn't entirely clear to me.

Reply:
It is meant that something like terraform import isn't supported. Essentially if you already have a cluster setup k0sctl won't be able to detect and do any operations with it. It is more or less just a "disclamer".

Reply:
Then I would phrase it as:

```
k0sctl is currently unable to manage existing resources, I.E, clusters installed manually, or with external tools. This means, k0sctl cannot:
 - detect the number of cluster nodes
 - manage upgrades(?)
 - < any other outcome that might happen >
 ```

Reply:
> k0sctl is currently unable to manage existing resources

This is no entirely correct. It can manage them just fine as long as they mostly look like ones that k0sctl itself would have installed, meaning they were installed via `k0s install`.

Maybe something along the lines of:

```
k0sctl will not perform any discovery of hosts, it only operates on the hosts listed in the provided configuration
```